Anlab.Mvc/Controllers/AdminController.cs

```diff
@@ -2,24 +2,228 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using Anlab.Core.Domain;
 using AnlabMvc.Data;
+using AnlabMvc.Models.Order;
+using AnlabMvc.Models.Roles;
+using AnlabMvc.Models.User;
+using AnlabMvc.Services;
 using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Identity.EntityFrameworkCore;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
 
 namespace AnlabMvc.Controllers
 {
-    [Authorize(Roles = "Admin")]
+    [Authorize(Roles = RoleCodes.Admin + "," + RoleCodes.User)]
     public class AdminController : ApplicationController
     {
         private readonly ApplicationDbContext _dbContext;
+        private readonly UserManager<User> _userManager;
+        private readonly RoleManager<IdentityRole> _roleManager;
+        private readonly IOrderService _orderService;
 
-        public AdminController(ApplicationDbContext dbContext)
+        public AdminController(ApplicationDbContext dbContext, UserManager<User> userManager, RoleManager<IdentityRole> roleManager, IOrderService orderService)
         {
             _dbContext = dbContext;
+            _userManager = userManager;
+            _roleManager = roleManager;
+            _orderService = orderService;
         }
-        public IActionResult Index()
+
+        [Authorize(Roles = RoleCodes.Admin)]
+        public async Task<IActionResult> Index()
+        {
+            var adminUsers = await _userManager.GetUsersInRoleAsync(RoleCodes.Admin);
+            var adminRole = _roleManager.Roles.Single(a => a.Name == RoleCodes.Admin).Id;
+            var userRole = _roleManager.Roles.Single(a => a.Name == RoleCodes.User).Id;
+
+            var users = _dbContext.Users.Include(a => a.Roles).ToList();
+            
+            var usersRoles = new List<UserRolesModel>();
+            foreach (var user in users)
+            {
+                var ur = new UserRolesModel();
+                ur.User = user;                
+                if (user.Roles.Any(a => a.RoleId == adminRole))
+                {
+                    ur.IsAdmin = true;
+                }
+                else
+                {
+                    ur.IsAdmin = false;
+                }
+                if (user.Roles.Any(a => a.RoleId == userRole))
+                {
+                    ur.IsUser = true;
+                }
+                else
+                {
+                    ur.IsUser = false;
+                }
+                usersRoles.Add(ur);
+            }           
+
+            return View(usersRoles);
+        }
+
+        public IActionResult ListNonAdminUsers()
+        {
+            var adminRole = _roleManager.Roles.Single(a => a.Name == RoleCodes.Admin).Id;
```

Comment:
@jSylvestre I do like `var adminRole = await _roleManager.FindByNameAsync("Admin");` a bit better than this way, but both work fine.